src/GoogleAdWordsEventForwarder.js

```diff
@@ -35,23 +35,53 @@
             labels,
             customAttributeMappings,
             reportingService,
-            eventQueue = [];
+            eventQueue = [],
+            gtagSiteId;
 
         self.name = name;
 
         function processEvent(event) {
             var reportEvent = false;
+            var sendEventFunction = function() {};
+            var generateEventFunction = function () {};
+            var conversionLabel;
+            var eventPayload;
 
             if (isInitialized) {
+                // First, process anything in the queue
+                processQueue(eventQueue);
+
                 try {
-                    if (event.EventDataType == MessageType.PageView) {
-                        reportEvent = logPageEvent(event, false);
+                    if (window.gtag && forwarderSettings.enableGtag == 'True') {
+                        sendEventFunction = sendGtagEvent;
+                        generateEventFunction = generateGtagEvent;
+                        generateCommerceEvent = generateGtagCommerceEvent;
+                    } else if (window.google_trackConversion) {
+                        sendEventFunction = sendLegacyEvent;
+                        generateEventFunction = generateAdwordsEvent;
+                        generateCommerceEvent = generateAdwordsCommerceEvent;
```

Comment:
I think it's a bit unclear to a future developer if we have generateGtagEvent, generateGtagCommerceEvent, sendGtagEvent...but then we have generateAdwordsEvent, generateAdwordsCommerceEvent, and sendLegacyEvent.  I know that the adwords event are legacy, but the naming can cause some confusion. I recommend calling it `sendAdwordsEvent`.  Perhaps put a comment above line 59 that says 
```
// window.google_trackConvesion is a legacy API
```
Thoughts?